lib/libcheribsdtest_dynamic/cheribsdtest_dynamic_fptr.c

```diff
@@ -44,3 +44,14 @@ void
 {
 	return (&cheribsdtest_dynamic_dummy_func);
 }
+
+/*
+ * This function must only be referenced as a scalar.
+ */
+void
+cheribsdtest_dynamic_dummy_func_scalar_only(void)
```

Comment:
Why do we need a new function?

Reply:
I’m aiming to test the ‘lazy-trampolines’ that come with default-wrapped function pointers. To do this you’d need to first resolve an integer address to the function, then resolve a capability, and make sure that a) they are equal and b) the latter is a tagged capability. (I haven’t actually tested (b) here because that would assume the existence of CHERI. To avoid using ifdefs maybe we can just call the function?)

The new function is needed to provide this ordering of symbol resolution.